<launch>
    <!-- Configure the IP address of the SP1 -->
    <arg name="sp1_address" value="192.168.10.10" />

    <!-- Configure the destination where the calibration will saved -->
    <arg name="calib_file" value="/tmp/nerian_calib.yaml" />

    <!-- Download the current camera calibration -->
    <node pkg="nerian_sp1" type="download_calibration.sh" args="$(arg sp1_address) $(arg calib_file)" name="download_calib" output="screen" />

    <!-- Configure SP1 operation mode to stereo matching -->
    <!-- <node pkg="nerian_sp1" type="config_sp1_stereo.sh" args="$(arg sp1_address)" name="config_sp1" output="screen" /> -->

    <!-- For debugging it is possible to configure a launch prefix -->
    <!-- <arg name="launchPrefix" value="gdb -ex 'catch throw' -ex run \-\-args" /> -->
    <!-- <arg name="launchPrefix" value="valgrind" /> -->
    <arg name="launchPrefix" value="" />

    <!-- Launch the SP1 node -->
    <node pkg="nerian_sp1" type="nerian_sp1_node" name="nerian_sp1_node" output="screen" launch-prefix="$(arg launchPrefix)">
        <param name="point_cloud_intensity_channel" type="bool" value="true" />
        <param name="ros_coordinate_system" type="bool" value="true" />

        <param name="color_code_disparity_map" type="bool" value="false" />
        <param name="color_code_legend" type="bool" value="true" />

        <param name="use_tcp" type="bool" value="false" />

        <param name="remote_host" type="string" value="0.0.0.0" />
        <param name="remote_port" type="string" value="7681" />

        <param name="local_host" type="string" value="0.0.0.0" />
        <param name="local_port" type="string" value="7681" />

        <param name="calibration_file" type="string" value="$(arg calib_file)"/>
        <param name="frame" type="string" value="map" />

        <param name="delay_execution" type="double" value="2" />
        <param name="max_depth" type="double" value="-1" />
    </node>
</launch>
